src/virtual-machines/source-2.js

```diff
@@ -1462,23 +1447,39 @@ function show_heap_value(address) {
     }
 }
 
-function show_array_value(address) {
+function is_denotable_value(addr) {
```

Comment:
"denotable" means names can refer to the value. The name then "denotes" the value. I don't think that's the way you use the word is correct. For example null should be a denotable value.